lib/printer.js

```diff
@@ -107,7 +107,7 @@ function Printer(originalOptions) {
 
     // Print the entire AST generically.
     function printGenerically(path) {
-        return genericPrint(path, options, printGenerically);
+        return printComments(path, p => genericPrint(path, options, printGenerically));
```

Comment:
Please use a non-arrow function expression here to fix tests in older Node versions. Also the parameter name should be `path` instead of `p`, since the `print` argument gets passed to other code by `printComments`, and we don't always want to be printing the original `path`.